features/netsocket/CellularInterface.h

```diff
@@ -0,0 +1,161 @@
+     *  Please check documentation of connect() for default behavior of APN settings.
+     *
+     *  @param apn      Access point name.
+     *  @param uname    Username (optional).
+     *  @param pwd      Password (optional).
+     */
+    virtual void set_credentials(const char *apn, const char *uname = 0,
+                                 const char *pwd = 0) = 0;
+
+    /** Set the plmn. PLMN controls to what network device registers.
+     *
+     *  @param plmn     user to force what network to register.
+     */
+    virtual void set_plmn(const char *plmn) = 0;
+
+    /** Set the PIN code for SIM card.
+     *
+     *  @param sim_pin      PIN for the SIM card.
+     */
+    virtual void set_sim_pin(const char *sim_pin) = 0;
+
+    /** Attempt to connect to a cellular network with a PIN and credentials.
+     *
+     *  @param sim_pin     PIN for the SIM card.
+     *  @param apn         Access point name (optional).
+     *  @param uname       Username (optional).
+     *  @param pwd         Password (optional).
+     *  @return            NSAPI_ERROR_OK on success, or negative error code on failure.
+     */
+    virtual nsapi_error_t connect(const char *sim_pin, const char *apn = 0,
+                                  const char *uname = 0,
+                                  const char *pwd = 0) = 0;
+
+    /** Attempt to connect to a cellular network.
+     *
+     *  If the SIM requires a PIN, and it is invalid or not set, NSAPI_ERROR_AUTH_ERROR is returned.
+     *
+     *  @return         NSAPI_ERROR_OK on success, or negative error code on failure.
+     */
+    virtual nsapi_error_t connect() = 0;
+
+    /** Stop the interface.
+     *
+     *  @return         NSAPI_ERROR_OK on success, or error code on failure.
+     */
+    virtual nsapi_error_t disconnect() = 0;
+
+    /** Check if the connection is currently established.
+     *
+     * @return `true` if the cellular module have successfully acquired a carrier and is
+     *         connected to an external packet data network using PPP, `false` otherwise.
+     */
+    virtual bool is_connected() = 0;
+
+    /** Get the local IP address.
+     *
+     *  @return         Null-terminated representation of the local IP address,
+     *                  or null if no IP address has been received.
+     */
+    virtual const char *get_ip_address() = 0;
+
+    /** Get the local network mask.
+     *
+     *  @return         Null-terminated representation of the local network mask,
+     *                  or null if no network mask has been received.
+     */
+    virtual const char *get_netmask() = 0;
+
+    /** Get the local gateways.
+     *
+     *  @return         Null-terminated representation of the local gateway,
+     *                  or null if no network mask has been received.
+     */
+    virtual const char *get_gateway() = 0;
+
+    /** @copydoc NetworkInterface::cellularBase
+     */
+    MBED_DEPRECATED_SINCE("mbed-os-5.12", "Migrated to CellularInterface")
```

Comment:
How can something new be already deprecated?

Did you copy&paste the code without removing the deprecation notices?

Reply:
Ah.. just noticed that this is just to maintain backward compatibility.